src/sample/View.java

```diff
@@ -99,51 +101,47 @@ public void changed(ObservableValue<? extends Boolean> observable, Boolean oldVa
 
     private void prepTheGame(Text helpText, CheckBox defaults, TextField textInput, Stage startGUI)
     {
+        startGUI.close();
         int gridSize;
-
+            // if defaults are checked then apply the default 5x5 size
             if (defaults.isSelected())
             {
                 textInput.clear();
                 gridSize = 5;
             }
             else {
                 try {
+                    // else provide the input as the size but with validation on failing to parse
+                    // the input to a number (so empty or letters)
                     gridSize = Integer.parseInt(textInput.getText());
```

Comment:
Is there a way we can limit the input to only be numbers so we don't have to do this parsing logic?

Reply:
https://stackoverflow.com/questions/7555564/what-is-the-recommended-way-to-make-a-numeric-textfield-in-javafx

I can look to implement this. If I understand correctly it sets up a listener to ignore non-numbers.

Reply:
resolved